modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/prototype/PrototypedAPITestcase.java

```diff
@@ -0,0 +1,240 @@
+
+        this.userMode = userMode;
+    }
+
+    @DataProvider
+    public static Object[][] userModeDataProvider() {
+
+        return new Object[][]{
+                new Object[]{TestUserMode.SUPER_TENANT_ADMIN},
+                new Object[]{TestUserMode.TENANT_ADMIN}
+        };
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws APIManagerIntegrationTestException, XPathExpressionException {
+
+        super.init(userMode);
+        apiProvider = user.getUserName();
+
+        String apiPrototypeEndpointPostfixUrl = "am/sample/pizzashack/v1/api/menu";
+        apiEndPointUrl = gatewayUrlsWrk.getWebAppURLHttp() + apiPrototypeEndpointPostfixUrl;
+    }
+
+    @Test(groups = {"wso2.am"}, description = "Create an API with a prototype endpoint and invoke")
+    public void testPrototypedAPIEndpoint() throws Exception {
+
+        String apiID;
+        apiName = "APIMPrototypedEndpointAPI1";
+        String apiContext = "pizzashack-prototype";
+        String apiTags = "pizza, order, pizza-menu";
+        String apiDescription = "Pizza API:Allows to manage pizza orders " +
+                "(create, update, retrieve orders)";
+
+        apiIdentifier = new APIIdentifier(apiProvider, apiName, apiVersion);
+        APIRequest apiRequest = new APIRequest(apiName, apiContext, new URL(apiEndPointUrl));
+        apiRequest.setVersion(apiVersion);
+        apiRequest.setDescription(apiDescription);
+        apiRequest.setTags(apiTags);
+        apiRequest.setVisibility(APIDTO.VisibilityEnum.PUBLIC.getValue());
+        apiRequest.setProvider(apiProvider);
+
+        HttpResponse addAPIResponse = restAPIPublisher.addAPI(apiRequest);
+        assertEquals(addAPIResponse.getResponseCode(), Response.Status.CREATED.getStatusCode(),
+                "Invalid Response Code");
+        apiID = addAPIResponse.getData();
+
+        //Deployed API as a Prototyped API & check the status
+        WorkflowResponseDTO lcChangeResponse = restAPIPublisher.changeAPILifeCycleStatus(apiID,
+                APILifeCycleAction.DEPLOY_AS_PROTOTYPE.getAction());
+
+        HttpResponse response = restAPIPublisher.getAPI(apiID);
+        Gson g = new Gson();
+        APIDTO apidto = g.fromJson(response.getData(), APIDTO.class);
+        String endPointString = "{\"implementation_status\":\"prototyped\",\"endpoint_type\":\"http\"," +
+                "\"production_endpoints\":{\"config\":null," +
+                "\"url\":\"" + apiEndPointUrl + "\"}," +
+                "\"sandbox_endpoints\":{\"config\":null,\"url\":\"" + apiEndPointUrl + "\"}}";
+
+        JSONParser parser = new JSONParser();
+        JSONObject endpoint = (JSONObject) parser.parse(endPointString);
+        apidto.setEndpointConfig(endpoint);
+
+        restAPIPublisher.updateAPI(apidto);
+
+        assertTrue(lcChangeResponse.getLifecycleState().getState().equals("Prototyped"),
+                apiName + "  status not updated as Prototyped");
+        Thread.sleep(15000);
```

Comment:
Shall we avoid these thread sleeps?

Reply:
@malinthaprasan these were already there in the existing prototype-related tests. I also observed the lifecycle change assertions failures without the waits. can we apply waitForAPIDeploymentSync for all these cases?

Reply:
This is done